website/docs/developer/research/research-bamboo.md

```diff
@@ -0,0 +1,164 @@
+
+```yaml
+version: 2
+plan:
+  key: ORCA
+plan-permissions:
+- users:
+  - rhassan
+  permissions:
+  - view
+  - edit
+  - build
+  - clone
+  - admin
+```
+- Docker- It is also possible to run a job inside a docker image rather than the bamboo agent if required. Check this [documentation](https://hub.docker.com/r/atlassian/bamboo-agent-base) for more information.
+
+```yaml
+test job name:
+  key: TJN
+  docker:
+    image: alpine
+```
+- Artifact- Artifacts are files created by a job build that can be used in other stages. An artifact definition can be created to specify which artifacts to keep and share from a build job. In the example below, an artifact definition named `test-artifact` is created under the job `test job` which is shared with other builds and deployments. Only the file matching the pattern .xml will be stored under artifact directory. 
+
+```yaml
+test job:
+  artifacts:
+  - name: test-artifact
+    location: artifact
+    pattern: '*.xml'
+    shared: true
+```
+Check artifact [documentation](https://confluence.atlassian.com/bamboo/configuring-a-job-s-build-artifacts-289277071.html) for more information.
+
+- Checkout- This is a task to check out a repository for use by just one job. In the example below, the job named `test job` has a checkout task that will checkout to the default repo defined in the plan before the tasks executes.
+
+```yaml
+test job:
+  tasks:
+  - checkout:
+      description: Checkout Default Repository
+```
+Check this [documentation](https://confluence.atlassian.com/bamboo0800/checking-out-code-1077778795.html) for more information.
+
+
+## Running Bamboo specs
+In order to run the YAML definition file, it has to be stored at `bamboo-specs/bamboo.yml` or `bamboo-specs/bamboo.yaml`under the repository specified. Check this [documentation](https://confluence.atlassian.com/bamboo/bamboo-yaml-specs-938844479.html) for more information.
+
+:::important
+Make sure the linked repository have permissions to create plans within given project in order to execute the YAML definition and create a plan. 
+:::
+
+In order to run this YAML definition from github repo, user has to setup `Repository-stored Specs` on Bamboo CI/CD [website](https://ci.earthdata.nasa.gov/build/admin/create/newSpecs.action). From the Bamboo dashboard, Choose `Specs`->`Set up Specs repository` and then select `Build project` as `ORCA` and `Repository host` as `ORCA repo`.
+The next step is to create a `Webhook` on the source repository by going to the settings option so that Bamboo knows about new commits. Webhooks allow Github repositories to communicate with Bamboo. A webhook has been created for ORCA project `https://ci.earthdata.nasa.gov/rest/api/latest/repository/scan?repositoryId=375849190` which needs to be copied to the `webhook` section in cumulus-orca repo. Furthur discussions are needed for this step with the whole team. 
```

Comment:
I had assumed that whenever a build is triggered it would pull the spec from the branch it is testing. Docs linked say that this is triggered on commit... Will it still be able to build for individual branches with changes to their individual specs?

Reply:
yes it can build on individual branches. By default, all branches will have the same spec file of their parent branch(develop or master) but that can also be changed by making some changes in the ci website.

Reply:
Are the webhooks needed in github if Bamboo is already polling for changes? We may need to submit a NASD ticket for them to configure webhooks for us since those are all admin settings in the NASA repos.

Reply:
when a new spec file is added for a new plan, Bamboo is not able to recognize the definition file. I had to go to ci website and manually poll the spec file. So I am guessing adding webhook will fix this issue. We should give this a try and see what happens.

Reply:
ok